### PR TITLE
doc(hmac): Make endian_swap / digest_swap consistent

### PR DESCRIPTION
**Pull request by eunchan**
This commit revises `CFG.endian_swap` to be consistent to
`CFG.digest_swap`.

To make it clear, an example is added to the document.

### CODE REVIEW

**Comment by eunchan** [comment]
@cindychip I didn't revise DV env yet. I will try to revise and merge into this PR.

**Comment by eunchan** [comment]
#13497 #12895

**Review by msfschaffner — APPROVED** [review]
Thanks for aligning this!

File: sw/device/lib/dif/dif_hmac_unittest.cc [hunk 1]
```diff
@@ -32,7 +32,7 @@ class HmacTest : public testing::Test, public mock_mmio::MmioTest {
   struct ConfigRegister {
     bool hmac_enable = false;
     bool sha_enable = true;
-    bool msg_little_endian = true;
+    bool msg_big_endian = false;
```
Comment (eunchan) [inline sw/device/lib/dif/dif_hmac_unittest.cc:35]: @doreis-lowrisc I rename the field to `big_endian` as the meaning of the field was reverted in this PR.

File: sw/device/silicon_creator/lib/drivers/hmac.c [hunk 2]
```diff
@@ -26,7 +26,7 @@ void hmac_sha256_init(void) {
   uint32_t reg = 0;
   reg = bitfield_bit32_write(reg, HMAC_CFG_DIGEST_SWAP_BIT, false);
   // Enable endian swap since our inputs are little-endian.
-  reg = bitfield_bit32_write(reg, HMAC_CFG_ENDIAN_SWAP_BIT, true);
+  reg = bitfield_bit32_write(reg, HMAC_CFG_ENDIAN_SWAP_BIT, false);
```
Comment (eunchan) [inline sw/device/silicon_creator/lib/drivers/hmac.c:29]: CC: @alphan

**Comment by msfschaffner** [comment]
@jadephilipoom can you give this a look also? I believe you initially raised this inconsistency in the CSRs as part of the D3 review.

**Review by jadephilipoom — APPROVED** [review]
Looks good to me, thanks for this change! Mind updating the cryptolib driver as well? Should be just a bitflip on this line:
https://github.com/lowRISC/opentitan/blob/b1f6dddfdf29165e8c5f62c1a630e8b88025db6d/sw/device/lib/crypto/drivers/hmac.c#L28

**Review by cindychip — APPROVED** [review]
lgtm

**Comment by eunchan** [comment]
image
I do not have CW310 to test the failure. Tried `//sw/device/tests/crypto:ecdsa_p256_verify_functest_sim_dv` and passed. Is this sufficient?

**Comment by jadephilipoom** [comment]
> I do not have CW310 to test the failure. Tried `//sw/device/tests/crypto:ecdsa_p256_verify_functest_sim_dv` and passed. Is this sufficient?

What about `//sw/device/tests/crypto:ecdsa_p256_verify_functest_sim_verilator`? That one should agree with the FPGA result.

File: sw/device/lib/crypto/drivers/hmac.c [hunk 3]
```diff
@@ -25,7 +25,7 @@ void hmac_sha256_init(void) {
   // Digest is little-endian by default.
   reg = bitfield_bit32_write(reg, HMAC_CFG_DIGEST_SWAP_BIT, false);
   // Enable endian swap since our inputs are little-endian.
```
Comment (jadephilipoom) [inline sw/device/lib/crypto/drivers/hmac.c:27]: This comment doesn't make sense any more, maybe:
```suggestion
  // Message is little-endian by default.
```

File: sw/device/silicon_creator/lib/drivers/hmac.c (hunk 2 above)
Comment (alphan) [reply sw/device/silicon_creator/lib/drivers/hmac.c:29]: LGTM, thanks!

**Pull request merged**